src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -109,7 +117,27 @@ private void checkIfUserExists(User userToBeCreated) {
                     String.format(baseErrorMessage, "username " + userByUsername,  "is"));
         }
     }
-
+    private void checkIfUserCanBeUpdated(User userToBeUpdated, User userInRepository) {
+        if (userInRepository == null){
+            String baseErrorMessage="User with Id %s not found";
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format(baseErrorMessage, userToBeUpdated.getId()));
+        }
+      
+          else if (userToBeUpdated.getUsername().equals("")){
+          String baseErrorMessage = "Username must not be empty!";
+          throw new ResponseStatusException(HttpStatus.CONFLICT, String.format(baseErrorMessage    ));
+          }
+          else if(!userToBeUpdated.getUsername().equals(userInRepository.getUsername()))
```

Comment:
nice check, first make sure that it is not his own username already. (there should be no error if he keeps the same username)